python/ray/serve/controller.py

```diff
@@ -62,8 +63,17 @@ def set_shadow(self, backend: str, proportion: float):
             self.shadow_dict[backend] = proportion
 
 
-BackendInfo = namedtuple("BackendInfo",
-                         ["worker_class", "backend_config", "replica_config"])
+class BackendInfo(BaseModel):
```

Comment:
Why is BackendInfo a pydantic class? We don't need to validate its input and it's purely for internal use. Can we use dataclasses instead?

Reply:
That makes sense, I'll change it to a dataclass. Dataclasses are in py3.7+ and have been backported to 3.6, so we would need to include dataclass as a pip dependency, is that okay?

Reply:
Yes! Let's add dataclasses as dependency for Serve